src/core/sliding_window_extremum.h

```diff
@@ -0,0 +1,112 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+    
+--*/
+
+#pragma once
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+typedef struct SLIDING_WINDOW_EXTREMUM_ENTRY {
+
+    uint64_t Value;
+
+    uint64_t Time;
```

Comment:
Units?

Reply:
This data structure is not designed for any specific kind of unit. You can use second, ms, us or even tick. In our use case, we use the packet number as "time".

I agree the `uint64_t Time` may not be named aptly. Any suggestions here?

Reply:
Do we use it with different units, or always the same? I agree it could be generic, but I'm not convinced we need to define it so.

Reply:
For a single instance of this "sliding window", the unit should be always the same.

But for different "windows", we can choose different units.